### ISSUE TITLE
[no-unused-vars] False positive with parameter decorators

### ISSUE THREAD

**Issue by raymondfeng** [typescript-eslint/typescript-eslint#571]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unused-vars": [
      "error",
      {
        "vars": "all",
        "args": "none"
      }
    ]
  }
}
```

```TS
// Create a parameter decorator
const log = function(target: Object, method: string, index: number) {
  console.log('decorating %s[%d]', method, index); 
}

class MyService {
  constructor(@log public x: string) {}
}
```

**Expected Result**

No violation of `unused-vars` is reported.

**Actual Result**

It complains that `log` is declared but not used.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.4.2` |
| `@typescript-eslint/parser`        | `1.4.2` |
| `TypeScript`                       | `3.4.5` |
| `ESLint`                           | `5.16.0` |
| `node`                             | `12.1.0` |
| `npm`                              | `6.9.0` |
